lib/safe_code/validator.ex

```diff
@@ -25,6 +25,18 @@ defmodule SafeCode.Validator do
     |> validate_quoted(opts)
   end
 
+  @spec validate_heex(binary, keyword) :: {:ok, ast :: Macro.t()} | {:error, %Phoenix.LiveView.HTMLTokenizer.ParseError{}}
```

Comment:
Giving a second thought, I think `validate_heex` should capture and return whatever exception raises to maintain the return consistent with the fn signature, eg: calling `Validator.validate_heex("System.cmd(...)")` should return `{:error, InvalidNode}`.

```suggestion
  @spec validate_heex(binary, keyword) :: {:ok, ast :: Macro.t()} | {:error, Exception.t()}
```

Wdyt?

Reply:
yeah, I agree. InvalidNode is another exception that we are expecting to throw, so it's better to catch 'em all